src/finite_differencing.jl

```diff
@@ -3,34 +3,34 @@ export check_Dv, check_Dv_update, check_errs
 """
     approximate_Dv(
         f,
-        ȳ::ArrayOr∇Real,
-        x::Tuple{Vararg{ArrayOr∇Real}},
-            v::Tuple{Vararg{ArrayOr∇Real}},
+        ȳ::∇ArrayOrScalar,
+        x::Tuple{Vararg{∇ArrayOrScalar}},
+            v::Tuple{Vararg{∇ArrayOrScalar}},
```

Comment:
Can we please un-indent this? It's definitely something I did mistakenly a while ago, might as well sort it now.

Reply:
Ah, sure! I'll take care of that and merge afterwards.